libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -288,13 +288,27 @@ void AP_ESC_Telem::update_telem_data(uint8_t esc_index, const AP_ESC_Telem_Backe
     }
 
     _telem_data[esc_index].count++;
+    if (_telem_data[esc_index].count == 0) {
+        // reset the CRC error counter each time the telemetry packet counter overflows,
+        // to prevent calculating CRC error rate with the incorrect (overflown) packet counter.
+        _telem_data[esc_index].CRC_error_counter = 0;
+    }
     _telem_data[esc_index].types |= data_mask;
     _telem_data[esc_index].last_update_ms = AP_HAL::millis();
 }
 
+// callback to increment the CRC error counter data in the frontend, should be called by the driver when a RX CRC error occurs
+void AP_ESC_Telem::increment_CRC_error_counter(const uint8_t esc_index)
```

Comment:
Does this not occur when trying to read telemetry information? Do we not need it per-ESC? Seems like it might be better to provide as part of the update*() APIs

Reply:
Yes when reading the telemetry information from the ESC, the CRC can missmatch. We need it per ESC so that a problematic ESC can be identified.

I do not think it should be a part of the update API, because in normal situation this will not happen often, hence the function only gets call on demand, and under normal usage it should almost never get called. So I do not think we should pass this on every single update*() API.